storage/src/main/scala/coop/rchain/storage/Key.scala

```diff
@@ -119,39 +120,17 @@ class Key(keyIn: String) extends TermTree with Ordered[Key] {
     bindings
   }
 
-  def compare(key: Key): Int = {
-    if (term == key.term) { return 0 }
-
-    if (arity > key.arity) { return 1 } else if (arity < key.arity) {
-      return -1
-    }
+  // Compare two keys in a manner analagous to strcmp(key.term, this.term)
+  // with these modifications:
+  //   sort by: arity, letter, digit
+  //
+  // Arity is sorted such that f(1) will be ahead of f(1,1).
+  // In unification, only keys of the same arity will be compared.
+  // Predicates will smaller arity are sorted first for reasons
+  // of readability in listing.
+  //
+  // Letters are sorted ahead of digits by fiat.
 
-    val length = math.min(term.length, key.term.length)
-    for (i <- 0 until length) {
-      if (term(i) != key.term(i)) {
-        if (Character.isLetter(term(i))
-            && Character.isDigit(key.term(i))) {
-          return -1
-        } else if (Character.isDigit(term(i))
-                   && Character.isLetter(key.term(i))) {
-          return 1
-        } else {
-          if (term(i) < key.term(i)) {
-            return -1
-          } else if (term(i) > key.term(i)) {
-            return 1
-          }
-        }
-      }
-    }
-
-    assert(term.length != key.term.length)
-    if (term.length > key.term.length) { return 1 }
-    -1
-  }
-
-  // sort by: arity, character, digit, other
-/*
   def compare(key: Key): Int = {
     var returnVal = 0
     var returnValSet = false
```

Comment:
Looks like this is unneeded, since returnValSet is only true when returnVal is nonzero.

Reply:
Removed